create calendar submit

namacosansan opened the pull request:
課題を提出いたします。 ご確認をお願いいたします。

namacosansan committed:
create calendar submit

namacosansan committed:
Delete blank lines

yuuu commented on 02.calendar/calendar1.rb:
```diff
@@ -0,0 +1,32 @@
+end
+
+command = ARGV.getopts("y:m:")
+year = command["y"] ? command["y"].to_i : Date.today.year
```
ここは次のように書くと少しだけスッキリしますね。

```ruby
year = (command["y"] || Date.today.year).to_i
```

namacosansan replied:
変数year,monthに関する記述を
year = (options["y"] || Date.today.year).to_i
month = (options["m"] || Date.today.month).to_i
に変更しました。

yuuu commented on 02.calendar/calendar1.rb:
```diff
@@ -0,0 +1,32 @@
+  print "\n" unless last_day.saturday? 
+end
+
+command = ARGV.getopts("y:m:")
```
`-y` や `-m` で指定された値は「コマンド」ではないので、別の名前を検討ください 🙏

namacosansan replied:
変数名をcommandからoptionsに変更しました。

yuuu commented on 02.calendar/calendar1.rb:
```diff
@@ -0,0 +1,32 @@
+  puts "日 月 火 水 木 金 土"
+
+  (first_day..last_day).each do |date|
+    print "   " * date.wday if date == first_day
```
繰り返し処理の1回目にしか行わない処理であれば、繰り返し処理に入る前に実行すれば `if` 式は不要となるはずです。

namacosansan replied:
繰り返し処理の前に１週目の処理を移動し、print "   " * first_day.wdayに修正しました。

yuuu commented on 02.calendar/calendar1.rb:
```diff
@@ -0,0 +1,32 @@
+  (first_day..last_day).each do |date|
+    print "   " * date.wday if date == first_day
+    if date.saturday? && date == today
+      print "\e[7m#{date.day.to_s.rjust(2)}\e[0m \n"
```
すべての分岐で `date.day.to_s.rjust(2)` という記述が存在するので、共通化できないか検討をお願いします 🙏

namacosansan replied:
day_str = date.day.to_s.rjust(2)とし、day_strを参照する形にしました。

yuuu commented on 02.calendar/calendar1.rb:
```diff
@@ -0,0 +1,32 @@
+
+  (first_day..last_day).each do |date|
+    print "   " * date.wday if date == first_day
+    if date.saturday? && date == today
```
「土曜日かどうか」と「今日かどうか」は組み合わせて判定する必要がありますか？組み合わせで判定すると現時点で4分岐になってしまいますし、今後何か条件が増えると倍々ゲームで記述する分岐数が増えていってしまいます。

別々に判定できれば、2回if式を書くだけで済むので記述量は減るはずです。

namacosansan replied:
”土曜日で改行する”条件文と”今日かどうか”の条件文を分割しました。

namacosansan committed:
- Fix description of first week of calendar
- Simplified date.day.to_s.rjust(2)
- Split conditional statements
- Changed variable names from “command” to “option"
- Changed conditional statement to logical OR
